### PR TITLE
2.3 release notes: Highlights, Doc and Bluetooth

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.3.rst [hunk 1]
```diff
@@ -9,7 +9,19 @@ We are pleased to announce the release of Zephyr RTOS version 2.3.0.
...
+* A new Devicetree API, based on hierarchical macros, has been introduced. This
+  new API allows the C code to access virtually all nodes and properties in a
+  clean, organized fashion
+* The kernel timeout API has been overhauled to be flexible and configurable,
+  with future support for features like 64-bit and absolute timeouts in mind
+* A new k_heap/sys_heap heap allocator has been introduced, which much better
+  performance than the existing k_mem_pool/sys_mem_pool
+* Zephyr now integrates with the TF-M (Trusted Firmware M) PSA-compliant
+  framework
+* The Bluetooth Low Energy Host now supports Advertisement Extensions
```
Comment (ioannisg) [inline doc/releases/release-notes-2.3.rst:23]: Advertising* Extensions ?

**Review by ioannisg — COMMENTED** [review]
just a comment about AE, otherwise looks ok

File: doc/releases/release-notes-2.3.rst (hunk 1 above)
Comment (carlescufi) [reply doc/releases/release-notes-2.3.rst]: Fixed

File: doc/releases/release-notes-2.3.rst [hunk 2]
```diff
@@ -430,15 +442,31 @@ Bluetooth
 
 * Host:
 
-  * <TBD>
+  * Support for Advertisement Extensions has been added.
+  * The Host is now 5.2 compliant, with support for EATT, L2CAP ECRED mode and
+    all new GATT PDUs.
+  * New user-controller data length and PHY update APIs.
```
[the remark this answers is not in the document]
Comment (carlescufi) [reply doc/releases/release-notes-2.3.rst]: Fixed

**Review by ioannisg — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

File: doc/releases/release-notes-2.3.rst [hunk 3]
```diff
@@ -9,7 +9,19 @@ We are pleased to announce the release of Zephyr RTOS version 2.3.0.
...
 
-* <TBD>
+* A new Zephyr CMake package has been introduced, reducing the need for
+  environment variables
+* A new Devicetree API, based on hierarchical macros, has been introduced. This
+  new API allows the C code to access virtually all nodes and properties in a
+  clean, organized fashion
+* The kernel timeout API has been overhauled to be flexible and configurable,
+  with future support for features like 64-bit and absolute timeouts in mind
+* A new k_heap/sys_heap heap allocator has been introduced, which much better
```
Comment (wopu-ot) [inline doc/releases/release-notes-2.3.rst:19]: which->with

**Review by wopu-ot — CHANGES_REQUESTED** [review]
Minor typo, otherwise looks fine to me.

File: doc/releases/release-notes-2.3.rst (hunk 3 above)
Comment (carlescufi) [reply doc/releases/release-notes-2.3.rst:19]: Fixed, thanks

File: doc/releases/release-notes-2.3.rst [hunk 4]
```diff
@@ -9,7 +9,19 @@ We are pleased to announce the release of Zephyr RTOS version 2.3.0.
...
 
-* <TBD>
+* A new Zephyr CMake package has been introduced, reducing the need for
+  environment variables
+* A new Devicetree API, based on hierarchical macros, has been introduced. This
+  new API allows the C code to access virtually all nodes and properties in a
+  clean, organized fashion
+* The kernel timeout API has been overhauled to be flexible and configurable,
+  with future support for features like 64-bit and absolute timeouts in mind
+* A new k_heap/sys_heap heap allocator has been introduced,with much better
```
Comment (wopu-ot) [inline doc/releases/release-notes-2.3.rst:19]: Sorry for being pedantic, but "introduced,with" should be "introduced, with" (space after comma).

**Review by wopu-ot — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.3.rst (hunk 4 above)
Comment (carlescufi) [reply doc/releases/release-notes-2.3.rst]: Thanks, fixed now :)

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**